system-test/error-reporting.ts

```diff
@@ -479,82 +487,100 @@ describe('error-reporting', () => {
     transport = new ErrorsApiTransport(configuration, logger);
   }
 
-  after(done => {
-    console.error = oldLogger;
+  after(async () => {
     if (transport) {
-      transport.deleteAllEvents(err => {
-        assert.ifError(err);
-        done();
-      });
+      await transport.deleteAllEvents();
     }
   });
 
   afterEach(() => {
     logOutput = '';
   });
 
-  function verifyAllGroups(
-      messageTest: (message: string) => void, timeout: number,
-      cb: (matchedErrors: ErrorGroupStats[]) => void) {
-    setTimeout(() => {
-      transport.getAllGroups((err, groups) => {
-        assert.ifError(err);
-        assert.ok(groups);
-
-        const matchedErrors = groups!.filter(errItem => {
-          return (
-              errItem && errItem.representative &&
-              errItem.representative.serviceContext &&
-              errItem.representative.serviceContext.service === SERVICE &&
-              errItem.representative.serviceContext.version === VERSION &&
-              messageTest(errItem.representative.message));
-        });
-
-        cb(matchedErrors);
+  async function verifyAllGroups(
+      messageTest: (message: string) => void, maxCount: number,
```

Comment:
I see `(message: string) => void` used in a few places.  Could we level that up to a type at the top and reuse it?